code/modules/mob/living/basic/farm_animals/bee/_bee.dm

```diff
@@ -0,0 +1,339 @@
+	. = ..()
+	if(gone != queen)
+		return
+	queen = null
+	// the bee should not exist without a bee.
+	if(!QDELETED(src))
+		qdel(src)
+
+/obj/item/queen_bee/proc/handle_needle(obj/item/source, obj/item/syringe, mob/living/user, params)
+	SIGNAL_HANDLER
+
+	if(!istype(syringe, /obj/item/reagent_containers/syringe))
+		return
+	var/obj/item/reagent_containers/syringe/needle = syringe
+	if(needle.reagents.has_reagent(/datum/reagent/royal_bee_jelly, 5)) //checked twice, because I really don't want royal bee jelly to be duped
+		needle.reagents.remove_reagent(/datum/reagent/royal_bee_jelly, 5)
+		var/obj/item/queen_bee/new_bee = new(get_turf(src))
+		new_bee.queen = new(new_bee)
+		if(queen?.beegent)
+			new_bee.queen.assign_reagent(queen.beegent) //Bees use the global singleton instances of reagents, so we don't need to worry about one bee being deleted and her copies losing their reagents.
+		user.put_in_active_hand(new_bee)
+		user.visible_message(span_notice("[user] injects [src] with royal bee jelly, causing it to split into two bees, MORE BEES!"),span_warning("You inject [src] with royal bee jelly, causing it to split into two bees, MORE BEES!"))
+		return
+	var/datum/reagent/chemical = needle.reagents.get_master_reagent()
+	if(chemical && needle.reagents.has_reagent(chemical.type, 5))
+		needle.reagents.remove_reagent(chemical.type, 5)
+		queen.assign_reagent(chemical)
+		user.visible_message(span_warning("[user] injects [src]'s genome with [chemical.name], mutating its DNA!"),span_warning("You inject [src]'s genome with [chemical.name], mutating its DNA!"))
+		name = queen.name
+	else
+		to_chat(user, span_warning("You don't have enough units of that chemical to modify the bee's DNA!"))
+
+/obj/item/queen_bee/suicide_act(mob/living/user)
+	user.visible_message(span_suicide("[user] eats [src]! It looks like [user.p_theyre()] trying to commit suicide!"))
+	user.say("IT'S HIP TO EAT BEES!")
+	qdel(src)
+	return TOXLOSS
+
+/obj/item/queen_bee/bought
+
+/obj/item/queen_bee/bought/Initialize(mapload)
+	. = ..()
+	queen = new(src)
+
+/obj/item/trash/bee
+	name = "bee"
+	desc = "No wonder the bees are dying out, you monster."
+	icon = 'icons/mob/simple/bees.dmi'
+	icon_state = "bee_item"
+	///the reagent the bee carry
+	var/datum/reagent/beegent
+	///the bee of this corpse
+	var/bee_type = /mob/living/basic/bee
+
+/obj/item/trash/bee/Initialize(mapload, mob/living/basic/bee/dead_bee)
+	. = ..()
+	if(dead_bee)
+		pixel_x = dead_bee.pixel_x
+		pixel_y = dead_bee.pixel_y
+		bee_type = dead_bee.type
+		if(dead_bee.beegent)
+			beegent = dead_bee.beegent
+			reagents.add_reagent(beegent.type, 5)
+		update_appearance()
+	AddComponent(/datum/component/edible, list(/datum/reagent/consumable/nutriment/vitamin = 5), null, BEE_FOODGROUPS, 10, 0, list("bee"), null, 10)
+	AddElement(/datum/element/swabable, CELL_LINE_TABLE_QUEEN_BEE, CELL_VIRUS_TABLE_GENERIC_MOB, 1, 5)
+	RegisterSignal(src, COMSIG_ATOM_ON_LAZARUS_INJECTOR, PROC_REF(use_lazarus))
```

Comment:
reorganize this entire proc to be an early return (so just do the whole addcomponent/element/signal stuff first, then early return on an `if(isnull(dead_bee))`